build/cmake/functions.cmake

```diff
@@ -97,11 +106,21 @@ function(wx_set_common_target_properties target_name)
     elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND NOT wxCOMMON_TARGET_PROPS_DEFAULT_WARNINGS)
         target_compile_options(${target_name} PRIVATE
             -Wall
+            -Wundef
+            -Wunused-parameter
+            -Wno-ctor-dtor-privacy
+            -Woverloaded-virtual
```

Comment:
Would it be worth adding `common_gcc_clang_warnings` variable to avoid duplicating this below? We could also add `-Wno-deprecated-declarations` to this variable if we had it.

Reply:
Good idea, changed to common variable